tests/system/action/mediafile/test_update.py

```diff
@@ -0,0 +1,267 @@
+        model = self.get_model("mediafile/111")
+        assert model.get("access_group_ids") == [3, 6]
+        assert model.get("inherited_access_group_ids") == [3, 6]
+        assert model.get("has_inherited_access_groups") is True
+
+    def test_update_parent_case4(self) -> None:
+        self.create_model("group/1", {"name": "group_LxAHErRs", "user_ids": []})
+        self.create_model("group/2", {"name": "group_sdfafd", "user_ids": []})
+        self.create_model("group/3", {"name": "group_ghjeei", "user_ids": []})
+        self.create_model(
+            "mediafile/110",
+            {
+                "title": "title_srtgb199",
+                "child_ids": [111],
+                "inherited_access_group_ids": [1, 2],
+                "access_group_ids": [1, 2],
+            },
+        )
+        self.create_model(
+            "mediafile/111", {"title": "title_srtgb123", "parent_id": 110},
+        )
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "mediafile.update",
+                    "data": [
+                        {
+                            "id": 111,
+                            "title": "title_Xcdfgee",
+                            "access_group_ids": [2, 3],
+                        }
+                    ],
+                }
+            ],
+        )
+        self.assertEqual(response.status_code, 200)
+        model = self.get_model("mediafile/111")
+        assert model.get("access_group_ids") == [2, 3]
+        assert model.get("inherited_access_group_ids") == [2]
+        assert model.get("has_inherited_access_groups") is True
+
+    def test_update_parent_case5(self) -> None:
+        self.create_model("group/1", {"name": "group_LxAHErRs", "user_ids": []})
+        self.create_model("group/2", {"name": "group_sdfafd", "user_ids": []})
+        self.create_model("group/3", {"name": "group_ghjeei", "user_ids": []})
+        self.create_model(
+            "mediafile/110",
+            {
+                "title": "title_srtgb199",
+                "child_ids": [111],
+                "inherited_access_group_ids": [1, 2],
+                "access_group_ids": [1, 2],
+            },
+        )
+        self.create_model(
+            "mediafile/111", {"title": "title_srtgb123", "parent_id": 110},
+        )
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "mediafile.update",
+                    "data": [
+                        {"id": 111, "title": "title_Xcdfgee", "access_group_ids": [3]}
+                    ],
+                }
+            ],
+        )
+        self.assertEqual(response.status_code, 200)
+        model = self.get_model("mediafile/111")
+        assert model.get("access_group_ids") == [3]
+        assert model.get("inherited_access_group_ids") == []
+        assert model.get("has_inherited_access_groups") is False
```

Comment:
This should be True not False

Reply:
Two tests were not correct. I fixed them and the code.